core/frontend/src/tile/RealityTileLoader.ts

```diff
@@ -242,3 +243,23 @@ export abstract class RealityTileLoader {
     return minDistance;
   }
 }
+
+/** Exposed strictly for testing purposes.
+* @internal
+*/
+export function createReaderPropsWithBaseUrl(streamBuffer: ByteStream | Uint8Array, yAxisUp: boolean, baseUrl?: string): GltfReaderProps | undefined {
+  if(streamBuffer instanceof ByteStream) {
+    return (!baseUrl || !isValidURL(baseUrl)) ? GltfReaderProps.create(streamBuffer.nextBytes(streamBuffer.arrayBuffer.byteLength), yAxisUp) : GltfReaderProps.create(streamBuffer.nextBytes(streamBuffer.arrayBuffer.byteLength), yAxisUp, new URL(baseUrl));
+  }
+  return (!baseUrl || !isValidURL(baseUrl)) ? GltfReaderProps.create(streamBuffer, yAxisUp) : GltfReaderProps.create(streamBuffer, yAxisUp, new URL(baseUrl));
+
+}
+
+function isValidURL(url: string){
+  try {
+    new URL(url);
+  } catch {
+    return false;
+  }
+  return true;
+}
```

Comment:
```suggestion
export function createReaderPropsWithBaseUrl(streamBuffer: ByteStream, yAxisUp: boolean, baseUrl?: string): GltfReaderProps | undefined {
  let url: URL | undefined;
  try {
    url = new URL(baseUrl);
  } catch (_) { }
  
  return GltfReaderProps.create(streamBuffer.nextBytes(streamBuffer.arrayBuffer.byteLength), yAxisUp, url);
```
You appear to accept `Uint8Array` strictly because your test wants to pass one in - make the test pass in a ByteStream.